### ISSUE TITLE
Paper comments

### ISSUE THREAD

**Issue by JoelForamitti** [ABM4ALL/Melodie#8]
This issue is part of the following review: https://github.com/openjournals/joss-reviews/issues/5100

Here are some comments on the paper:

- Statement of need
  - It would be helpful to not only describe how Melodie differs from Mesa and AgentPy, but also explain what need this satisfies. E.g. I understand that the environment component does in Melodie what is usually done in the Model component in Mesa and AgentPy, but what is the advantage of this design? The same applies to the second and third point.
  - To me it seems that a key novelty of this package is that is uses an sql database. Could you mention why you chose this approach and what the advantage is?
  - Please add a sentence on why you chose to work in Python, and mention that there are a large amount of ABM frameworks based on other programming languages.
- Overview
  - It is difficult to get an overview of how modelling in the package works by reading this section. I would suggest to show how an agent, a model, a secnario, and a visualizer is defined, the same way it is shown for the simulator.
  - Would it be possible to present a flowchart, showing how trainer, calibrator, scenario, etc. feed into each other?
  
Some typos:
- l12: mechanism -> mechanisms
- l15: ; (2) -> ; and (2)
- l57: data object -> data objects

**Comment by JoelForamitti** [comment]
Another novelty that you might want to highlight and explain the motivation is that this package uses cython components.

**Comment by SongminYu** [comment]
Hi @JoelForamitti, thanks a lot for your review and comments! We improve the paper accordingly and come back to you :)

**Comment by SongminYu** [comment]
Hi @JoelForamitti, thanks for your valuable comments and suggestions. I have updated and pushed the manuscript to the JOSS-paper branch. The “Statement of Need” section is enhanced according to your suggestions. However, for the two overview points, I haven't successfully made the changes at the moment. 

First, due to the length limit, we didn't include the details about how to set up each class. But this is introduced in detail in our documentation: (1) the tutorial section, and (2) the model gallery. We provided a link to the documentation and mentioned this in the manuscript. 

Second, I also really wanted to add a figure to show how the modules are linked together. However, the figures I can think of are either too complicated and difficult to read or too simple and a bit misleading, especially when the trainer and the calibrator are also included. I tried to explain this in the tutorial in detail, as well as the documentation for each model in the gallery.
